core/src/server/rpc_module.rs

```diff
@@ -738,6 +740,90 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 	}
 }
 
+/// Represent a pending subscription which waits being accepted or rejected.
+///
+/// Warning: you need to call either `PendingSubscription::accept` or `PendingSubscription::reject` otherwise
+/// the subscription will not make any progress.
+#[derive(Debug)]
+struct InnerPendingSubscription {
+	/// Sink.
+	sink: MethodSink,
+	/// Get notified when subscribers leave so we can exit
+	close_notify: Option<Arc<Notify>>,
+	/// MethodCallback.
+	method: &'static str,
+	/// Unique subscription.
+	uniq_sub: SubscriptionKey,
+	/// Shared Mutex of subscriptions
+	subscribers: Subscribers,
+	/// Request ID.
+	id: Id<'static>,
+}
+
+/// Represent a pending subscription which waits for being accepted or rejected.
+///
+/// This type implement drop for ease of use, such as dropped in error short circuiting via `map_err()?`.
+#[derive(Debug)]
+pub struct PendingSubscription(Option<InnerPendingSubscription>);
+
+impl PendingSubscription {
+	/// Reject the subscription call.
+	pub fn reject(self, err: Error) -> Result<(), Error> {
+		self.reject_from_error_object(err.as_error_object())
+	}
+
+	/// Reject the subscription call from [`ErrorObject`].
+	pub fn reject_from_error_object(mut self, err: ErrorObject) -> Result<(), Error> {
```

Comment:
since `subscription.close()` doesn't expose a way to work with `ErrorObject`, I wonder whether we should be consistent and not expose it here either?

Reply:
it is because the `Error type` is not clone I used it. It was needed by the proc macro code